lib/fog/vsphere/requests/compute/vm_clone.rb

```diff
@@ -569,9 +573,11 @@ def vm_clone(options = {})
                                                                       :pool => resource_pool,
                                                                       :diskMoveType => :moveChildMostDiskBacking)
           else
-            relocation_spec = RbVmomi::VIM.VirtualMachineRelocateSpec(:datastore => datastore_obj,
-                                                                      :pool => resource_pool,
+            relocation_spec = RbVmomi::VIM.VirtualMachineRelocateSpec(:pool => resource_pool,
                                                                       :transform => options['transform'] || 'sparse')
+            unless options.key?('storage_pod')
+              relocation_spec[:datastore] = datastore_obj unless datastore_obj.nil?
```

Comment:
Could you check both conditionals in one `unless` only?
It would make it easier to follow whats happening.